heartbeat/mdraid

```diff
@@ -260,28 +268,21 @@ array_is_linear_or_raid0() {
 	[ $c -eq `echo $devs|wc -w` ] && return 0 || return 1
 }
 
-# Return true for clustered RAID relying on all component devices reporting clustered type;
-# presumes defined global devs variable with the component devices of the array.
+# Return true for clustered RAID
 array_is_clustered_raid() {
-	local c=0 d dev_count=`echo $devs|wc -w` s
 	# Check based on specific "intent bitmap : clustered" output once
 	# available in mdadm output or fall back to "Cluster Name" defined
 	# presuming bitmap is clustered if so.
 	local strs="clustered cluster.name"
 
-	for d in $devs
-	do
-		for s in $strs
-		do
-			$MDADM -E $d 2>&1 | grep -i "$s" >/dev/null 2>&1
-			if [ $? -eq 0 ]; then
-				c=$((c+1))
-				break
-			fi
-		done
-	done
-
-	[ $c -eq $dev_count ] && return 0 || return 1
+	
+	$MDADM -D $md_dev 2>&1 | $EGREP -i "$s" >/dev/null 2>&1
```

Comment:
You probabbly meant to search for one of the strings in `$strs` here.

Reply:
yes, in my patch, only "clustered" is valid. it is only work with mdadm upstream commit 9e4494051de3f53228fabae56c116879bff5a0c8

Reply:
I got your meaning. it is my fault, should be "$strs".

Reply:
after analysis, the original code is correct.  I am sorry to file this useless patch.
I found my env had some problems. blkid showed there were 4 disks(sda sdb sde sdf) had same uuid. but wipefs only show 2 disks (sda sdb) had raid info.

```
[tb-clustermd1 ~]# lsblk                                                                                  
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT                                                                
sda      8:0    0  300M  0 disk                                                                           
sdb      8:16   0  300M  0 disk                                                                           
sdc      8:32   0  300M  0 disk                                                                           
sdd      8:48   0  300M  0 disk                                                                           
sde      8:64   0  300M  0 disk                                                                           
sdf      8:80   0  300M  0 disk                                                                           
vda    253:0    0   40G  0 disk                                                                           
├─vda1 253:1    0    8M  0 part                                                                           
├─vda2 253:2    0   38G  0 part /                                                                         
└─vda3 253:3    0    2G  0 part [SWAP]                                                                    
[tb-clustermd1 ~]# blkid                                                                                  
/dev/vda2: UUID="1ced82f1-240f-4375-8506-8e67e8342368" BLOCK_SIZE="4096" TYPE="ext4" PARTUUID="53e10073-9d
c5-44c1-86cd-bd15e24232fd"                                                                                
/dev/vda3: UUID="0cca454f-c44b-4489-83dc-14f86ffec51c" TYPE="swap" PARTUUID="5bef0e53-e87b-455d-b52f-d4f15
45942ac"                                                                                                  
/dev/sdb: UUID="581ff564-f118-f567-5ea7-16ce23202a73" UUID_SUB="24959508-da73-674d-d04d-5cefc973eceb" LABE
L="tb-clustermd1:0" TYPE="linux_raid_member"                                                              
/dev/sda: UUID="581ff564-f118-f567-5ea7-16ce23202a73" UUID_SUB="367538a8-0d07-bde8-050e-4d3afe250da3" LABE
L="tb-clustermd1:0" TYPE="linux_raid_member"                                                              
/dev/sde: UUID="581ff564-f118-f567-5ea7-16ce23202a73" UUID_SUB="367538a8-0d07-bde8-050e-4d3afe250da3" LABE
L="tb-clustermd1:0" TYPE="linux_raid_member"                                                              
/dev/sdf: UUID="581ff564-f118-f567-5ea7-16ce23202a73" UUID_SUB="367538a8-0d07-bde8-050e-4d3afe250da3" LABE
L="tb-clustermd1:0" TYPE="linux_raid_member"                                                              
/dev/vda1: PARTUUID="393fa0a0-3a03-4ca4-9729-24c66208aa79"                                                
[tb-clustermd1 ~]# wipefs /dev/sda                                                                        
DEVICE OFFSET TYPE              UUID                                 LABEL                                
sda    0x1000 linux_raid_member 581ff564-f118-f567-5ea7-16ce23202a73 tb-clustermd1:0                      
[tb-clustermd1 ~]# wipefs /dev/sdb
DEVICE OFFSET TYPE              UUID                                 LABEL
sdb    0x1000 linux_raid_member 581ff564-f118-f567-5ea7-16ce23202a73 tb-clustermd1:0
[tb-clustermd1 ~]# wipefs /dev/sde <== empty output
[tb-clustermd1 ~]# wipefs /dev/sdf  <== empty output
```

I didn't know where introduced the blkid issue. after dd zero to sde sdf, I tested many time, the mdraid RA worked fine.